contracts/eosio.system/src/rex.cpp

```diff
@@ -677,14 +695,11 @@ namespace eosiosystem {
             auto& return_buckets = rb.return_buckets;
             auto iter = return_buckets.begin();
             while ( iter != return_buckets.end() && iter->first <= time_threshold ) {
-               auto next = iter;
-               ++next;
                const uint32_t overtime = get_elapsed_intervals( effective_time,
                                                                 iter->first + seconds(rex_return_pool::total_intervals * rex_return_pool::dist_interval) );
                surplus      += iter->second * overtime;
                expired_rate += iter->second;
-               return_buckets.erase(iter);
-               iter = next;
+               iter         =  return_buckets.erase(iter);
```

Comment:
Better yet delay the erase until the end:

```cpp
auto iter = return_buckets.begin();
for (; iter != return_buckets.end() && iter->first <= time_threshold; ++iter) {
   const uint32_t overtime = get_elapsed_intervals(effective_time,
                                                   iter->first + seconds(rex_return_pool::total_intervals * rex_return_pool::dist_interval));

   surplus      += iter->second * overtime;
   expired_rate += iter->second;
}
return_buckets.erase(return_buckets.begin(), iter);
```

Reply:
Will use it.